### PR TITLE
Toppview refactor

### PR DESCRIPTION
**Pull request by cbielow**
# Description

[FEATURE] caches the spectra tree view (does not repaint it), because this could make selecting spectra in 1D view extremely slow (usually when there are a lot(!) of spectra/chromatograms, as is the case for OpenSwath XIC data).

[FEATURE] displaying chromatogram data with a lot of chroms is now feasible by making a quadratic precursor mapping function linear

[FEATURE] support loading sqMass sqlLite files from TOPPView and visualize Chromatograms

[FIXES] #4942 (Layer insertion/deletion failures), by introducing a LayerStack class which encapsulates the problem and gets rid of a lot of boilerplate code

[MISC] some other fixes, e.g. getting rid of std::map in FileTypes by using static arrays and having a compile-time type description available for all file types we can read/write. Useful for file dialogs and filters.

# Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes (GUI so not applicable, unfortunately)
- [x] Updated or added python bindings for changed or new classes (GUI, so not applicable)

### CODE REVIEW

**Comment by timosachsenberg** [comment]
just testing if the rebase action now works

**Comment by cbielow** [comment]
rebuild jenkins

**Comment by cbielow** [comment]
ready for review

File: src/openms_gui/source/VISUAL/SpectraSelectionTabs.cpp [hunk 1]
```diff
@@ -103,9 +103,9 @@ namespace OpenMS
     });
 
     SpectrumCanvas* cc = tv_->getActiveCanvas();
-    int layer_row = (cc == nullptr ? -1 : (int)cc->activeLayerIndex());
+    Size layer_row = (cc == nullptr ? -1 : cc->getCurrentLayerIndex());
 
-    if (cc == nullptr || layer_row == -1)
+    if (cc == nullptr || layer_row == (Size)-1)
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/SpectraSelectionTabs.cpp:108]: is this test redundant?
e.g., test only for layer_row == (Size)-1 should suffice

File: src/openms_gui/include/OpenMS/VISUAL/SpectrumCanvas.h [hunk 2]
```diff
@@ -62,6 +62,110 @@ namespace OpenMS
 {
   class SpectrumWidget;
 
+  class LayerStack
```
Comment (timosachsenberg) [inline src/openms_gui/include/OpenMS/VISUAL/SpectrumCanvas.h:65]: move implementation to cpp file.
maybe add a line of comment that layers are automatically sorted by layer type?

File: src/openms_gui/include/OpenMS/VISUAL/LayerData.h [hunk 3]
```diff
@@ -394,6 +405,40 @@ namespace OpenMS
     */
     void updateRanges();
 
+    /// Returns the minimum intensity of the internal data, depending on type
+    inline float getMinIntensity() const
```
Comment (timosachsenberg) [inline src/openms_gui/include/OpenMS/VISUAL/LayerData.h:409]: move to cpp and remove inline?

**Comment by timosachsenberg** [comment]
nice fixes and refactoring. Otherwise looks good (already had a look at it yesterday)

File: src/openms_gui/source/VISUAL/SpectraSelectionTabs.cpp (hunk 1 above)
Comment (cbielow) [reply src/openms_gui/source/VISUAL/SpectraSelectionTabs.cpp:108]: good point. looking a bit closer, layer_row is not required at all..

Comment (cbielow) [reply src/openms_gui/source/VISUAL/SpectraSelectionTabs.cpp]: looking even closer, it is...

**Pull request merged**